### DISCUSSION TITLE
How to hide the mouse cursor

### DISCUSSION THREAD

**Discussion by brabebhin** [microsoft/WindowsAppSDK#3601] [Q&A] [answered]
I've been trying to wrap my head around this one: how to hide the mouse cursor? this is pretty straigthforward on both UWP and win32 but winUI 3 seems to have no way of doing this.

**Comment by DarranRowe** [comment]
To put it simply, WinUI 3 applications are desktop (Windows API/Win32) applications. WinUI 3 is basically a library/framework, therefore it needs an existing process model to work on. So use the Windows API method of hiding the cursor.

**Reply by brabebhin** [reply]
Yes, i suspected as much, however, I'd wanted a nice way of doing it with winui3 APIs

**Comment by castorix** [comment, marked as answer]
It can be done with "_set_ProtectedCursor_"  (System.Reflection)
A test by disposing the cursor of "Test" Button to hide it when I click on "Click" Button : 

![Hide_Cursor](https://user-images.githubusercontent.com/22345506/235376788-3f655647-0282-4db2-96ba-99202b1489d5.gif)

**Reply by brabebhin** [reply]
Wouldn't that be the ProtectedCursor that's available in UI elements? What value are you setting?

**Reply by castorix** [reply]
> Wouldn't that be the ProtectedCursor that's available in UI elements? What value are you setting?

I just set the basic Microsoft.UI.Input.InputSystemCursorShape.Arrow cursor
Then I called Dispose() to destroy/hide it (it was the idea of someone else in another thread...)
(can also be done with an empty cursor (just transparent pixels), but needs more code)

**Reply by brabebhin** [reply]
Great idea with the transparent pixels, this really never crossed my mind. It may actually fit my use case better than the Dispose approach.
